money-core/src/main/scala/com/comcast/money/core/internal/SpanLocal.scala

```diff
@@ -46,35 +46,32 @@ object SpanLocal extends SpanContext {
   override def current: Option[Span] = Option(threadLocalCtx.get).flatMap(_.headOption)
 
   override def push(span: Span): Unit =
-    if (span != null)
+    if (span != null) {
       Option(threadLocalCtx.get) match {
-        case Some(stack) =>
-          stack.push(span)
-          setSpanMDC(Some(span.info.id))
-          setSpanNameMDC(Some(span.info.name))
+        case Some(current) =>
+          threadLocalCtx.set(span :: current)
         case None =>
-          threadLocalCtx.set(new Stack[Span]())
-          push(span)
+          threadLocalCtx.set(List(span))
       }
+      setSpanMDC(Some(span.info.id))
+      setSpanNameMDC(Some(span.info.name))
+    }
 
   override def pop(): Option[Span] =
-    Option(threadLocalCtx.get).map {
-      stack =>
-        // remove the current span in scope for this thread
-        val spanId = stack.pop()
-
-        // rolls back the mdc span to the parent if present, if none then it will be cleared
-        val head = stack.headOption
+    threadLocalCtx.get match {
```

Comment:
interesting that you dropped the `Option` here, I think technically it is ok but I don't like pattern matching on a null

Reply:
I am ok leaving this as is, I did some reading and testing and it seems like what you had was ok.  Having an initial value always there is cool